analyzers/tests/SonarAnalyzer.Test/TestCases/RouteTemplateShouldNotStartWithSlash.AspNetCore.cs

```diff
@@ -0,0 +1,259 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Routing;
+
+[Route("[controller]")]
+public class NoncompliantController : Controller // Noncompliant {{Change the paths of the actions of this controller to be relative and adapt the controller route accordingly.}}
+//           ^^^^^^^^^^^^^^^^^^^^^^
+{
+    [Route("/Index1")]                  // Secondary
+//   ^^^^^^^^^^^^^^^^
+    public IActionResult Index1() => View();
+
+    [Route("/SubPath/Index2")]          // Secondary
+//   ^^^^^^^^^^^^^^^^^^^^^^^^
+    public IActionResult Index2() => View();
+
+    [HttpGet("/[action]")]              // Secondary
+//   ^^^^^^^^^^^^^^^^^^^^
+    public IActionResult Index3() => View();
+
+    [HttpGet("/SubPath/Index4_1")]      // Secondary
+//   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    [HttpGet("/[controller]/Index4_2")] // Secondary
+//   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    public IActionResult Index4() => View();
+}
+
+[Route("[controller]")]
+[Route("[controller]/[action]")]
+public class NoncompliantMultiRouteController : Controller // Noncompliant {{Change the paths of the actions of this controller to be relative and adapt the controller route accordingly.}}
+//           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+{
+    [Route("/Index1")]                  // Secondary
+//   ^^^^^^^^^^^^^^^^
+    public IActionResult Index1() => View();
+
+    [Route("/SubPath/Index2")]          // Secondary
+//   ^^^^^^^^^^^^^^^^^^^^^^^^
+    public IActionResult Index2() => View();
+
+    [HttpGet("/[action]")]              // Secondary
+//   ^^^^^^^^^^^^^^^^^^^^
+    public IActionResult Index3() => View();
+
+    [HttpGet("/SubPath/Index4_1")]      // Secondary
+//   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    [HttpGet("/[controller]/Index4_2")] // Secondary
+//   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    public IActionResult Index4() => View();
+}
+
+[Route("[controller]")]
+public class CompliantController : Controller // Compliant: at least one action has at least a relative route
+{
+    [Route("/Index1")]
+    public IActionResult Index1() => View();
+
+    [Route("/SubPath/Index2")]
+    public IActionResult Index2() => View();
+
+    [HttpGet("/[action]")]
+    public IActionResult Index3() => View();
+
+    [HttpGet("/[controller]/Index4_1")]
+    [HttpGet("SubPath/Index4_2")] // The relative route
+    public IActionResult Index4() => View();
+}
+
+public class NoncompliantNoControllerRouteController : Controller // Noncompliant {{Change the paths of the actions of this controller to be relative and add a controller route with the common prefix.}}
```

Comment:
This contradicts suggestion 3) for rule S6934. This is not necessarily a bad thing as option 3) is still not ideal. But if you want to do route value binding for a single action only, it is a viable alternative. You just have to accept the S6931 violation afterward. I think we should cross-link between the two RSpecs.
Please look at the next test case as well. We allow absolute path in this rule as well if there is mixed routing.
/cc @costin-zaharia-sonarsource and @antonioaversa